omnibus/python-scripts/packages.py

```diff
@@ -186,7 +222,9 @@ def install_diff_packages_file(install_directory, filename, exclude_filename):
         if install_package_name in exclude_packages:
             print(f"Skipping '{install_package_name}' as it's already included in '{exclude_filename}' file")
         else:
-            if install_package_line.startswith('datadog-'):
+            # install_package_line can be a <requirement specifier> which contains a package name and a version constraint
+            # we need to check that install_package_line doesn't contain any dependency name in DEPS_STARTING_WITH_DATADOG
+            if install_package_line.startswith('datadog-') and not any(install_package_line.startswith(dep) for dep in DEPS_STARTING_WITH_DATADOG):
```

Comment:
do we need to be more precise than `startswith` ? e.g. `datadog-log` will match any name that `datadog-logger` matches., and `datadog-sma` could match a future (for example) integration named `datadog-smart-devices`.

Reply:
Oh, nice catch! Yes we definitely need to! We need to properly extract the dependency name from the install_package_line and then check if it exists in the list.
Thanks @clarkb7 !

Reply:
Fixed!

Reply:
thanks!